src/models/Teams.php

```diff
@@ -60,14 +60,15 @@ public function __construct(public Users $Users, ?int $id = null)
      */
     public function getTeamsFromIdOrNameOrOrgidArray(array $input): array
     {
+        $Config = Config::getConfig();
         $res = array();
         $sql = 'SELECT id, name FROM teams WHERE id = :query OR name = :query OR orgid = :query';
         $req = $this->Db->prepare($sql);
         foreach ($input as $query) {
             $req->bindParam(':query', $query);
             $this->Db->execute($req);
             $team = $req->fetch();
-            if ($team === false) {
+            if ($team === false && $Config->configArr['saml_team_create']) {
```

Comment:
the call to `createTeamIfAllowed()` does the same check. What your change does is avoid an exception being thrown if the team cannot be created. I'll probably need to test this further to ensure it does what we want it to do. The code could also probably benefit from some refactoring.

Reply:
I'm happy to refactor/simplify this further. This is the only place `createTeamIfAllowed` is called, so one could rename the function to `createTeam` and remove the check (and exception) from there entirely. There are also two other places `getTeamsFromIdOrNameOrOrgidArray` is called from. However, I'm not yet familiar with the codebase, so yes, there might be side effects now, where the code generated an exception previously and is now failing with an uncought error.